var/spack/repos/builtin/packages/postgresql/package.py

```diff
@@ -125,3 +125,22 @@ def setup_dependent_run_environment(self, env, dependent_spec):
             env.prepend_path('TCLLIBPATH', self.prefix.lib)
         if '+python' in spec:
             env.prepend_path('PYTHONPATH', self.prefix.lib)
+
+    @property
+    def libs(self):
+        stat_libs = ['libecpg_compat', 'libecpg', 'libpgcommon',
+                     'libpgcommon_shlib', 'libpgfeutils', 'libpgport',
+                     'libpgport_shlib', 'libpgtypes', 'libpq']
+        fl_stat = find_libraries(stat_libs, self.prefix, shared=False,
+                                 recursive=True)
```

Comment:
Can you either avoid recursion or start in a sub-folder of `prefix`? Coding something like the above might be an issue if somebody adds an external for `postgresql` with `prefix='/usr'` :slightly_smiling_face:

Reply:
`find_libraries` avoids recursion intelligently by first checking in `prefix.lib` and `prefix.lib64` without recursion and only checking in `prefix` recursively if it isn't found already. So this is the right approach, and what we do in every other package.

Reply:
Hmm, I remember people posting issues about that, but maybe it was a custom "root-like" location used at some specific site?